core/src/setup/Kokkos_Setup_Cuda.hpp

```diff
@@ -104,7 +104,8 @@
 #if !defined(KOKKOS_DISABLE_CUDA_ASM)
 #if !defined(KOKKOS_ENABLE_CUDA_ASM)
 #define KOKKOS_ENABLE_CUDA_ASM
-#if !defined(KOKKOS_DISABLE_CUDA_ASM_ATOMICS)
+#if !defined(KOKKOS_DISABLE_CUDA_ASM_ATOMICS) && \
+    defined(KOKKOS_ENABLE_GNU_ATOMICS)
```

Comment:
Can you elaborate on this change and explain what was wrong?

Reply:
CUDA_ASM uses GNU Atomics macros for memory orders and so we can't do this in windows with clang.